### PR TITLE
Lzk/partitioning

### PR DESCRIPTION
**Pull request by lzianekhodja**
Static partitioning

### CODE REVIEW

File: infrastructure/modules/armonik/compute-plane-hpa.tf [hunk 1]
```diff
@@ -1,14 +1,14 @@
 resource "helm_release" "keda_hpa_compute_plane" {
-  count      = length(var.compute_plane)
-  name       = "hpa-${var.compute_plane[count.index].name}"
+  for_each   = toset(local.partition_names)
+  name       = "${each.key}-compute-plane"
```
Comment (lemaitre-aneo) [inline infrastructure/modules/armonik/compute-plane-hpa.tf:3]: `"compute-plane-${each.key}"`?

File: infrastructure/modules/armonik/compute-plane.tf [hunk 2]
```diff
@@ -43,40 +46,34 @@ resource "kubernetes_deployment" "compute_plane" {
...
-              limits   = {
-                cpu    = var.compute_plane[count.index].polling_agent.limits.cpu
-                memory = var.compute_plane[count.index].polling_agent.limits.memory
-              }
-              requests = {
-                cpu    = var.compute_plane[count.index].polling_agent.requests.cpu
-                memory = var.compute_plane[count.index].polling_agent.requests.memory
-              }
+          resources {
+            limits   = {
```
Comment (lemaitre-aneo) [inline infrastructure/modules/armonik/compute-plane.tf:70]: I think you could write this with the same effect:
```
limits = var.compute_plane[each.key].polling_agent.limits
```

File: infrastructure/modules/armonik/locals.tf [hunk 3]
```diff
@@ -10,13 +13,13 @@ locals {
   admin_gui_node_selector_values = values(local.admin_gui_node_selector)
 
   # Node selector for compute plane
-  compute_plane_node_selector        = [for index in range(0, length(var.compute_plane)) : try(var.compute_plane[index].node_selector, {})]
-  compute_plane_node_selector_keys   = [for index in range(0, length(local.compute_plane_node_selector)) : keys(local.compute_plane_node_selector[index])]
-  compute_plane_node_selector_values = [for index in range(0, length(local.compute_plane_node_selector)) : values(local.compute_plane_node_selector[index])]
+  compute_plane_node_selector        = {for partition in local.partition_names : partition => try(var.compute_plane[partition].node_selector, {})}
```
Comment (lemaitre-aneo) [inline infrastructure/modules/armonik/locals.tf:16]: You can write the loop like this:
```terraform
{for partition, compute_plane in var.compute_plane : parition => try(compute_plane.node_selector, {})}
```

**Review by lemaitre-aneo — COMMENTED** [review]
I would recommend to have only 2 partitions in the example deployments.

You can maybe add a variable for the control plane to have a default partition.

File: infrastructure/modules/armonik/compute-plane-hpa.tf (hunk 1 above)
Comment (lzianekhodja) [reply infrastructure/modules/armonik/compute-plane-hpa.tf]: why not

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**